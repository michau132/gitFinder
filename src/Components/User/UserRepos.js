import React from 'react';
import PropTypes from 'prop-types';
import UserRepoListItem from './UserRepoListItem';

const UserRepos = ({ userRepos }) => (
  <ul className="container__userInfoRepoList">
    <h4>Repositiories:</h4>
    {
      userRepos.map(listItem => (
        <UserRepoListItem listItem={listItem} key={listItem.id} />
      ))
    }
  </ul>
);

UserRepos.propTypes = {
  userRepos: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      pushed_at: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default UserRepos;
