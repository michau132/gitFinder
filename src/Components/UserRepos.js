import React from 'react';
import PropTypes from 'prop-types';
import UserReposListItem from './UserReposListItem';

const UserRepos = ({ userRepos }) => (
  <ul>
    <h4>Repositiories:</h4>
    {
      userRepos.map(listItem => (
        <UserReposListItem listItem={listItem} key={listItem.id} />
      ))
    }
  </ul>
);

UserRepos.propTypes = {
  userRepos: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      pushed_at: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default UserRepos;
