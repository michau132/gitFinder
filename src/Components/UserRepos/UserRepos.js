import React from 'react';
import PropTypes from 'prop-types';
import styles from './style.css';
import UserReposListItem from '../UserReposListItem/UserReposListItem';

const UserRepos = ({ userRepos }) => (
  <div className={styles.userRepos}>
    <div className={styles.filterForm}>
      <span className={styles.square} />
      <form className={styles.form}>
        <input className={styles.formText} type="text" placeholder="filter projects" />
        <button type="submit">Open</button>
        <button type="submit">Hide</button>
        <button type="submit">Show all</button>
      </form>
    </div>
    <fieldset>
      <legend>Projects</legend>
      <ul className={styles.listRepos}>
        {
            userRepos.map(listItem => (
              <UserReposListItem listItem={listItem} key={listItem.id} />
            ))
          }
      </ul>
    </fieldset>
  </div>
);

UserRepos.propTypes = {
  userRepos: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      pushed_at: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default UserRepos;
