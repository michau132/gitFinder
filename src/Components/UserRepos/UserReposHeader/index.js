
import React from 'react';
import PropTypes from 'prop-types';
import styles from './style.css';
import Button from '../../Button';

const UserReposHeader = ({
  setReposOnKeyUp,
  openSelectedRepos,
  hideSelectedRepos,
  showAllRepos,
  selectAllRepos,
  allReposAreShown,
  selectedReposAreEmpty,
  filterProjectsInput,
  allReposAreSelected,
}) => (
  <div className={styles.filterForm}>
    <div className={styles.checkAll}>
      <span>Select all</span>
      <input type="checkbox" onChange={selectAllRepos} checked={allReposAreSelected} />
    </div>
    <form className={styles.form}>
      <input className={styles.formText} value={filterProjectsInput} type="text" placeholder="filter projects" onChange={e => setReposOnKeyUp(e.target.value)} />
      <Button
        type="button"
        onClick={openSelectedRepos}
        disabled={selectedReposAreEmpty}
        text="open"
      />
      <Button
        type="button"
        onClick={hideSelectedRepos}
        disabled={selectedReposAreEmpty}
        text="hide"
      />
      <Button
        type="button"
        onClick={showAllRepos}
        disabled={allReposAreShown}
        text="show all"
      />
    </form>
  </div>
);

UserReposHeader.propTypes = {
  setReposOnKeyUp: PropTypes.func.isRequired,
  showAllRepos: PropTypes.func.isRequired,
  allReposAreShown: PropTypes.bool.isRequired,
  selectedReposAreEmpty: PropTypes.bool.isRequired,
  filterProjectsInput: PropTypes.string.isRequired,
  allReposAreSelected: PropTypes.bool.isRequired,
  openSelectedRepos: PropTypes.func.isRequired,
  hideSelectedRepos: PropTypes.func.isRequired,
  selectAllRepos: PropTypes.func.isRequired,
};

export default UserReposHeader;
