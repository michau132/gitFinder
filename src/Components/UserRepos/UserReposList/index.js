import React from 'react';
import PropTypes from 'prop-types';
import styles from './style.css';
import UserReposListItem from './UserReposListItem';

const UserReposList = ({
  userRepos,
  selectUserRepo,
  hideSingleRepo,
  openSingleRepo,
}) => (
  <fieldset>
    <legend>Projects</legend>
    <ul className={styles.listRepos}>
      {
        userRepos.map(listItem => (
          <UserReposListItem
            listItem={listItem}
            key={listItem.id}
            selectUserRepo={selectUserRepo}
            hideSingleRepo={hideSingleRepo}
            openSingleRepo={openSingleRepo}
          />
        ))
      }
    </ul>
  </fieldset>
);

UserReposList.propTypes = {
  userRepos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      forks_count: PropTypes.number,
      stargazers_count: PropTypes.number,
      description: PropTypes.string,
      html_url: PropTypes.string.isRequired,
    }),
  ).isRequired,
  hideSingleRepo: PropTypes.func.isRequired,
  openSingleRepo: PropTypes.func.isRequired,
  selectUserRepo: PropTypes.func.isRequired,
};

export default UserReposList;
