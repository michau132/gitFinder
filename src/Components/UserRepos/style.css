fieldset {
  padding: 0;
}
.userRepos {
  overflow: hidden; 
  max-height: inherit;
  max-width: 80%;
}

.filterForm {
  display: flex;
  margin-bottom: 15px;
}

.listRepos {
  overflow: hidden;
  overflow-y: scroll;
  position: relative;
  max-height: 500px;
  list-style: none;
}

.square {
  height: 13px;
  width: 13px;
  border: 1px solid black;
  align-self: center;
  margin: 0 20px;
  position: relative;
  display: inline-block;
}

.square:before {
  content: '';
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 9px;
  height: 1px;
  background-color: black;
}

.form {
  display: inline-block;
  width: calc(100% - 50px);
  display: flex;
  justify-content: space-between;
}

.formText {
  flex-grow: 1;
}

.form button {
  margin: 0 5px;
}

.form button:last-of-type {
  margin-right: 0;
}