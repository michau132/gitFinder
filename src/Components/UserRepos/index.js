import React from 'react';
import PropTypes from 'prop-types';
import UserReposList from './UserReposList';
import UserReposHeader from './UserReposHeader';
import styles from './style.css';

const UserRepos = ({
  userRepos,
  selectUserRepo,
  hideSingleRepo,
  openSingleRepo,
  allReposAreShown,
  ...restProps
}) => (
  <div className={styles.userRepos}>
    <UserReposHeader
      allReposAreShown={allReposAreShown}
      {...restProps}
    />
    <UserReposList
      userRepos={userRepos}
      selectUserRepo={selectUserRepo}
      hideSingleRepo={hideSingleRepo}
      openSingleRepo={openSingleRepo}
      allReposAreShown={allReposAreShown}
    />
  </div>
);

UserRepos.propTypes = {
  userRepos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      forks_count: PropTypes.number,
      stargazers_count: PropTypes.number,
      description: PropTypes.string,
      html_url: PropTypes.string.isRequired,
    }),
  ).isRequired,
  selectUserRepo: PropTypes.func.isRequired,
  hideSingleRepo: PropTypes.func.isRequired,
  openSingleRepo: PropTypes.func.isRequired,
};

export default UserRepos;
