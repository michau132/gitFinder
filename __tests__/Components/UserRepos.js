import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserRepos from '../../src/Components/UserRepos';

describe('testing UserRepos component', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      userRepos: [
        {
          id: 1,
          name: 'firstRepo',
          forks_count: 0,
          stargazers_count: 0,
          description: 'First repo description',
          html_url: 'https://github.com/aOko123/firstRepo',
        },
        {
          id: 2,
          name: 'secondRepo',
          forks_count: 0,
          stargazers_count: 0,
          description: 'Second repo description',
          html_url: 'https://github.com/aOko123/secondRepo',
        },
      ],
      selectUserRepo: jest.fn(),
      hideSingleRepo: jest.fn(),
      openSingleRepo: jest.fn(),
      setReposOnKeyUp: jest.fn(),
      openSelectedRepos: jest.fn(),
      hideSelectedRepos: jest.fn(),
      showAllRepos: jest.fn(),
      selectAllRepos: jest.fn(),
      allReposAreShown: true,
      selectedReposAreEmpty: true,
      filterProjectsInput: 'project',
      allReposAreSelected: false,
    };

    wrapper = shallow(<UserRepos {...props} />);
  });

  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });
  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
