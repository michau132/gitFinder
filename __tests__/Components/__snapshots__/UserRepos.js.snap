// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing UserRepos component matching snapshot 1`] = `
<div>
  <UserReposHeader
    allReposAreSelected={false}
    allReposAreShown={true}
    filterProjectsInput="project"
    hideSelectedRepos={[MockFunction]}
    openSelectedRepos={[MockFunction]}
    selectAllRepos={[MockFunction]}
    selectedReposAreEmpty={true}
    setReposOnKeyUp={[MockFunction]}
    showAllRepos={[MockFunction]}
  />
  <UserReposList
    allReposAreShown={true}
    hideSingleRepo={[MockFunction]}
    openSingleRepo={[MockFunction]}
    selectUserRepo={[MockFunction]}
    userRepos={
      Array [
        Object {
          "description": "First repo description",
          "forks_count": 0,
          "html_url": "https://github.com/aOko123/firstRepo",
          "id": 1,
          "name": "firstRepo",
          "stargazers_count": 0,
        },
        Object {
          "description": "Second repo description",
          "forks_count": 0,
          "html_url": "https://github.com/aOko123/secondRepo",
          "id": 2,
          "name": "secondRepo",
          "stargazers_count": 0,
        },
      ]
    }
  />
</div>
`;
